src/main/java/com/webdev/cosmo/cosmobackend/security/FacebookAuthentication.java

```diff
@@ -0,0 +1,58 @@
+package com.webdev.cosmo.cosmobackend.security;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.Setter;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+
+import java.util.Collection;
+
+
+@Getter
+@Setter
+@AllArgsConstructor
+public class FacebookAuthentication implements Authentication {
+
+    private boolean authenticated = false;
+    private FacebookUser facebookUser;
+
+    public FacebookAuthentication(FacebookUser facebookUser) {
+        this.facebookUser = facebookUser;
```

Comment:
Will this object change? Is it muttable?

Reply:
Another thing is, we do not want to store whole object here, FacebookUser is simply a contract between Us and Facebook. ExternalModel should not be stored and used later internally. Pick fields which You think may come in handy, define them here, and later define mapper which will map external object to internal.